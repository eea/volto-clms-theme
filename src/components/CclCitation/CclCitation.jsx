import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

/**
 * Citation component doc.
 * @function CclCitation
 * @param {string} title Header of Citation.
 * @param {boolean} marginBottom Default: false.
 * @param {node} children Add any html element.
 * @example <CclCitation title="Validation status">
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Duis luctus
          mauris ante, a iaculis leo placerat quis.
        </p>
      </CclCitation>
 * 
 */
function CclCitation({ title, marginBottom = false, children }) {
  let citationClass = classNames('validation-citation-container', {
    'citation-container': !marginBottom,
    'validation-container': marginBottom,
  });

  return (
    <div className={citationClass}>
      <div className="citation-title">{title}</div>
      {children}
    </div>
  );
}

CclCitation.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  marginBottom: PropTypes.bool,
};
export default CclCitation;
