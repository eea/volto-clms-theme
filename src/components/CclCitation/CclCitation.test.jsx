import React from 'react';
import renderer from 'react-test-renderer';
import CclCitation from './CclCitation';

test('Renders', () => {
  const component = renderer.create(
    <CclCitation title="test">children text</CclCitation>,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
