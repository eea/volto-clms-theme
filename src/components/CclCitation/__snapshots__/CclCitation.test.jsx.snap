// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders 1`] = `
<div
  className="validation-citation-container citation-container"
>
  <div
    className="citation-title"
  >
    test
  </div>
  children text
</div>
`;
